File: server/src/Bananagrams.Dal/Models/GameUser.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;

namespace Bananagrams.Dal.Models;

[Table("game_users")]
public class GameUser
{
    [Key]
    [Column("id")]
    public int Id { get; set; }
    [Column("game_id")]
    public int GameId { get; set; }
    [Column("user_id")]
    public int UserId { get; set; }
    public Game? Game { get; set; }
    public User? User { get; set; }
    public List<GameUserGameAnagram>? GameUserGameAnagrams { get; set; }
}

File: server.java/bananagrams/src/main/java/coe/bananagrams/entity/GameUser.java
package coe.bananagrams.entity;

import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;
import java.util.Set;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "game_users")
public class GameUser {
    @Id
    @Column(name = "id", updatable = false, nullable = false)
    private int id;
    @Column(name = "game_id", insertable = false, updatable = false, nullable = false)
    private int gameId;
    @Column(name = "user_id", insertable = false, updatable = false, nullable = false)
    private int userId;

    @ManyToOne
    private Game game;
    @ManyToOne
    private User user;
    @OneToMany(mappedBy = "gameUser")
    private Set<GameUserGameAnagram> gameUserGameAnagrams;
}
